### ISSUE TITLE
[Bug] AnalyzeDataFlow puts parameters only used with 'nameof()' in ReadInside

### ISSUE THREAD

**Issue by jamesqo** [dotnet/roslyn#21210]
I'm writing an analyzer for https://github.com/xunit/xunit.analyzers to help warn when a parameter to a `[Theory]` method is unused. I'm doing this by calling `SemanticModel.AnalyzeDataFlow` on the method, then checking which of the parameter symbols are not contained in `flowAnalysis.ReadInside`. I surprised to see that my analyzer doesn't trigger on this snippet:

```cs
using System;
using Xunit;

class TestClass
{
    [Theory]
    void TestMethod(int unused)
    {
        Console.WriteLine(nameof(unused));
    }
}
```

The value of `unused` isn't being read at all, yet `unused` is still in `ReadInside`.

**Comment by gafter** [comment]
No, `nameof` doesn't read any expression. It just yields a literal string.

**Comment by jamesqo** [comment]
@gafter Exactly, so why does `nameof` cause a variable to be included in `ReadInside`?

**Comment by gafter** [comment]
If it behaves as you say, that would be a bug.

**Comment by jamesqo** [comment]
@gafter I updated the issue title. Please feel free to reopen & change labels.
